internal/listener/listener.go

```diff
@@ -220,3 +221,32 @@ func (l *Listener) DumpIncidents(w http.ResponseWriter, r *http.Request) {
 	enc.SetIndent("", "  ")
 	_ = enc.Encode(encodedIncidents)
 }
+
+func (l *Listener) DumpSchedules(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		_, _ = fmt.Fprintln(w, "GET required")
+		return
+	}
+
+	if !l.checkDebugPassword(w, r) {
+		return
+	}
+
+	l.runtimeConfig.RLock()
+	defer l.runtimeConfig.RUnlock()
+
+	for _, schedule := range l.runtimeConfig.Schedules {
+		fmt.Fprintf(w, "[id=%d] %q:\n", schedule.ID, schedule.Name)
+
+		// Iterate in 30 minute steps as this is the granularity Icinga Notifications Web allows in the configuration.
+		// Truncation to seconds happens only for a more readable output.
+		step := 30 * time.Minute
+		start := time.Now().Truncate(time.Second)
```

Comment:
I did the `Truncate(Second)` to remove the nanosecond precision and the monotonic time from the output in a way where the time shown still exactly matches the time used for evaluation (so I'm not only specifying an output format with limited precision). Truncating to minutes would make little difference in that regard.

```go
package main

import (
	"fmt"
	"time"
)

func main() {
	t := time.Now()

	fmt.Printf("%v\n", t)
	// 2024-06-06 10:49:12.322749534 +0200 CEST m=+0.000012964

	fmt.Printf("%v\n", t.Truncate(time.Second))
	// 2024-06-06 10:49:12 +0200 CEST

	fmt.Printf("%v\n", t.Truncate(time.Minute))
	// 2024-06-06 10:49:00 +0200 CEST
}
```

Editing rotations in Notifications Web still feels weird for me. As of me trying it, it's 10:55 and I'm editing a rotation that's active at that very moment, it sets `actual_handoff` to 11:00, i.e. if I save the currently rotation get disabled until a few minutes in the future. I'd expect it to set `actual_handoff` to now instead, so that it becomes immediately active.

My code was written with that expectation in mind because if it's set to know, you see your change active after a second (assuming clocks are in sync) instead of once the next full minute is reached (or even next half hour if I'd truncate to that).